synapse/storage/databases/main/room.py

```diff
@@ -1396,8 +1396,6 @@ def set_room_is_public_appservice_txn(txn, next_id):
                     txn,
                     table="appservice_room_list",
                     keyvalues={
-                        "appservice_id": appservice_id,
-                        "network_id": network_id,
```

Comment:
Won't this mean that if we delete a room in one appservice list, it will delete them in all? While it's probably unexpected for a room to appear in multiple lists, its not wrong per se.

Should we instead be making the API require specifying the appservice ID for when done as an admin?

Reply:
Indeed, it appears multiple application services can specify the same protocol (`network_id`).

The problem is that the admin may not initially know which network the room belongs to (and I'm not sure what the best way to query that information is).

I might advocate for the other direction and just state that if an admin uses the endpoint, the room is removed across all appservices that are part of that network.

Reply:
I was assuming that the client would know the network ID as it would just have used it to do the `/publicRooms` query, but it doesn't when getting "all rooms".

> I might advocate for the other direction and just state that if an admin uses the endpoint, the room is removed across all appservices that are part of that network.

If we're doing that then maybe we should always delete the room across all room lists?

Reply:
By "all room lists" you mean the client public room list as well, right?

Would it be better to just have an Admin API to delete a room across both client and appservice room lists?

Otherwise I'm not sure how this PR will work without different requirements depending on the requester (which feels weird).

Reply:
> Would it be better to just have an Admin API to delete a room across both client and appservice room lists?

Yeah, that might make more sense, and is probably what we want? 

Separately, its not unreasonable for someone to want to delete entries from a particular room list, so it would be good to not make an API that designs that out.